webapp/src/cloud.ts

```diff
@@ -418,6 +476,80 @@ export async function convertCloudToLocal(userId: string) {
     }
 }
 
+const CLOUDSAVE_DEBOUNCE_MS = 3000;
+const CLOUDSAVE_MAX_MS = 15000;
```

Comment:
@eanders-MS , I re-worked the auto-save system entirely. Now we listen for "header:*" virtual API changes and then debounce them between 3-15 seconds.

I yanked out all the many places in the cloud that were making explicit cloud saves. Now instead the only ways to trigger a cloud save is via the virtual APIs or the syncAsync function.

One of the problems we had before is that we had to trust that callsites were calling "saveToCloud" at appropriate times and for good reason. And of course this wasn't true, it was being both overcalled in many places and also not called in important scenarios.

Reply:
Nice cleanup.